lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1435,13 +1435,23 @@ static LogicalResult canonicalizeSingleSetConnect(ConnectOp op,
 
   // Ok, we know we are doing the transformation.
 
-  // Make sure the constant dominates all users.
-  if (srcValueOp && srcValueOp != &declBlock->front())
+  // Replace with constant zero.
+  auto replacement = op.src();
+  if (isa_and_nonnull<InvalidValueOp>(srcValueOp)) {
```

Comment:
I don't know if there is a clean way to restructure the code but we're checking `srcValueOp != nullptr` 3 times in a row (`if (!srcValueOp) {` above)

Reply:
I can at least simplify the checks here with one level of nesting. I'll add this in a force push.